test/e2e/flask/btc/btc-send.spec.ts

```diff
@@ -0,0 +1,164 @@
+import { strict as assert } from 'assert';
+import { Suite } from 'mocha';
+import { Driver } from '../../webdriver/driver';
+import { DEFAULT_BTC_ACCOUNT, DEFAULT_BTC_BALANCE } from '../../constants';
+import {
+  getTransactionRequest,
+  SendFlowPlaceHolders,
+  withBtcAccountSnap,
+} from './common-btc';
+
+export async function startSendFlow(driver: Driver, recipient?: string) {
+  // Wait a bit so the MultichainRatesController is able to fetch BTC -> USD rates.
+  await driver.delay(1000);
+
+  // Start the send flow.
+  const sendButton = await driver.waitForSelector({
+    text: 'Send',
+    tag: 'button',
+    css: '[data-testid="coin-overview-send"]',
+  });
+
+  // Firefox test is flaky without this delay. The send flow doesn't start properly.
+  if (driver.browser === 'firefox') {
+    await driver.delay(1000);
+  }
+
+  await sendButton.click();
+
+  // See the review button is disabled by default.
+  await driver.waitForSelector({
+    text: 'Review',
+    tag: 'button',
+    css: '[disabled]',
+  });
+
+  if (recipient) {
+    // Set the recipient address (if any).
+    await driver.pasteIntoField(
+      `input[placeholder="${SendFlowPlaceHolders.RECIPIENT}"]`,
+      recipient,
+    );
+  }
+}
+
+describe('BTC Account - Send', function (this: Suite) {
+  it('can send complete the send flow', async function () {
+    await withBtcAccountSnap(
+      { title: this.test?.fullTitle() },
+      async (driver, mockServer) => {
+        await startSendFlow(driver, DEFAULT_BTC_ACCOUNT);
+
+        // TODO: Remove delay here. There is a race condition if the amount and address are set too fast.
```

Comment:
Quick question, where should the race condition be fixed? Would it be on the Snap side?

Reply:
It would be on the snap side, i think the new context update should fix this.